Composer/packages/client/src/pages/publish/BotStatusList.tsx

```diff
@@ -0,0 +1,325 @@
+// Copyright (c) Microsoft Corporation.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import moment from 'moment';
+import formatMessage from 'format-message';
+import { Checkbox } from 'office-ui-fabric-react/lib/Checkbox';
+import { Icon } from 'office-ui-fabric-react/lib/Icon';
+import React, { useState } from 'react';
+import { Fragment } from 'react';
+import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
+import { Spinner, SpinnerSize } from 'office-ui-fabric-react/lib/Spinner';
+import { PublishTarget } from '@bfc/shared';
+import {
+  CheckboxVisibility,
+  DetailsList,
+  IColumn,
+  IDetailsColumnStyleProps,
+  IDetailsColumnStyles,
+} from 'office-ui-fabric-react/lib/DetailsList';
+import { IStyleFunctionOrObject } from 'office-ui-fabric-react/lib/Utilities';
+import { IconButton } from 'office-ui-fabric-react/lib/Button';
+
+import { navigateTo } from '../../utils/navigation';
+
+import { IStatus, PublishStatusList } from './PublishStatusList';
+import { detailList, listRoot, tableView } from './styles';
+
+// Licensed under the MIT License.
+export type IBotStatus = {
+  id: string;
+  name: string;
+  publishTargets?: PublishTarget[];
+  publishTarget?: string;
+  time?: string;
+  status?: number;
+  message?: string;
+  comment?: string;
+};
+export type IBotStatusListProps = {
+  projectId: string;
+  items: IBotStatus[];
+  botPublishHistoryList: { projectId: string; publishHistory: IStatus[] }[];
+  updateItems: (items: IBotStatus[]) => void;
+  updatePublishHistory: (items: IStatus[], item: IBotStatus) => void;
+  updateSelectedBots: (items: IBotStatus[]) => void;
+  changePublishTarget: (PublishTarget: string, item: IBotStatus) => void;
+  onLogClick: (item: IStatus) => void;
+  onRollbackClick: (selectedVersion: IStatus, item: IBotStatus) => void;
+};
+
+export const BotStatusList: React.FC<IBotStatusListProps> = (props) => {
+  const {
+    projectId,
+    items,
+    botPublishHistoryList,
+    updateItems,
+    updatePublishHistory,
+    changePublishTarget,
+    updateSelectedBots,
+    onLogClick,
+    onRollbackClick,
+  } = props;
+  const [selectedBots, setSelectedBots] = useState<IBotStatus[]>([]);
+  const [showHistoryBots, setShowHistoryBots] = useState<string[]>([]);
+
+  const [currentSort, setSort] = useState({ key: 'Bot', descending: true });
+  const sortByName = (ev: React.MouseEvent<HTMLElement>, column: IColumn): void => {
```

Comment:
why do we need to write custom sort functionality on a DetailsList. those functionalities are built in.

Reply:
The default sort can not sort the list.